src/bsp/include/bsp/SocketContainer.h

```diff
@@ -0,0 +1,22 @@
+#ifndef __SOCKETCONTAINER_H_
+#define __SOCKETCONTAINER_H_
+
+#include "ITCPClient.h"
+#include "TCPClientWrapper.h"
+#include <logger/ILogger.h>
+#include <optional>
+
+/**@brief Containes the sockets for the application*/
+namespace SocketContainer {
+    /**
+     * @brief Return an instance of the platform dependent BSP.
+     *
+     * @b Warning note that if you make multiple request, you will obtaine multiple TCPClientWrapper
+     * objects, but under the hood, it's the same socket reference
+     */
+    std::optional<TCPClientWrapper> getHostClientSocket(const char* address,
+                                                        int port,
```

Comment:
I would maybe change specify the size of the `int` (we are compiling on 64 bit linux and 32 bit STM32)